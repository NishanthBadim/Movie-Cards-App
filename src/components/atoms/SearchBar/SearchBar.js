import React from 'react';
import '../../../index.css'

  
const SearchBar =(props) => {
    return(
    <div className="form-group has-search">
        <span className="fa fa-search form-control-feedback"></span>
        <input style={{display:'inline-block'}}  type="text" className="form-control" size="40" value={props.value} onChange={props.changed} placeholder="Search by Movie Name..." /> 
        <p className={props.result==='No Results Found' ? 'text-results' : ''} style={{display:'inline-block',fontSize:'18px',color:'gray'}}>{props.result}</p>
    </div>
    );
}
export default SearchBar;