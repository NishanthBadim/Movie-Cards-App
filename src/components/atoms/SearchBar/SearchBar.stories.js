import React from 'react'
import { withKnobs, text } from '@storybook/addon-knobs/react'
import { action } from '@storybook/addon-actions'
import './../../../index.css'

import SearchBar from './SearchBar'

export default {
    component: SearchBar,
    title: "SearchBar",
    decorators: [withKnobs]
}
export const Default = () => {
    return (
        <div className="form-group has-search">
            <span className="fa fa-search form-control-feedback"></span>
            <SearchBar value={text("value", "")} result={text("results", "")}
                changed={action("Entered Input")} />
        </div>
    )
}
export const MultipleResults = () => {
    return (
        <SearchBar value={text("value", "")} result={text("results", " 2 Results Found")}
            changed={action("Entered Input")} />
    )
}

export const SingleResult = () => {
    return (
        <SearchBar value={text("value", "")} result={text("results", " 1 Result Found")}
            changed={action("Entered Input")} />
    )
}

export const NoResults = () => {
    return (
        <SearchBar value={text("value", "")} result={text("results", "No Results Found")}
            changed={action("Entered Input")} />
    )
}