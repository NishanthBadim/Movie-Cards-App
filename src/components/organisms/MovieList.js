import React, { Component } from 'react';
import MovieCard from '../molecules/MovieCard';
import SearchBar from '../atoms/SearchBar/SearchBar';

export default class MovieList extends Component {
    constructor() {
        super();
        this.state = {
            search: ''
        }
    }

    filteredMovies = [];

    updateSearch = (event) => {
        this.setState({ search: event.target.value });
    }

    render() {
        this.filteredMovies = this.props.movies.filter(
            (movie) => {
                return movie.title.toLowerCase().indexOf(this.state.search.toLowerCase()) !== -1;
            }
        );
        let result = '';
        if (this.filteredMovies.length === 0) {
            result = 'No Results Found';
        }
        else if (this.filteredMovies.length === 1) {
            result = '1 Result Found';
        }
        else if (this.filteredMovies.length !== this.props.movies.length) {
            result = this.filteredMovies.length + ' Results Found';
        }

        return (
            <div>
                <SearchBar value={this.state.search} changed={this.updateSearch} result={result} />

                <div className="card-deck">
                    {
                        this.filteredMovies.map(movie => <MovieCard key={movie.id} movie={movie} />)
                    }
                </div>
            </div>
        )
    }
}

