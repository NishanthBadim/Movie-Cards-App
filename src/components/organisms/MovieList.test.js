import React from 'react';
import {create} from 'react-test-renderer'
import "@testing-library/jest-dom/extend-expect";
import MovieList from './MovieList'
import { __esModule } from 'react-modal-video';
import { fireEvent, render } from '@testing-library/react';
import SearchBar from '../atoms/SearchBar/SearchBar';

const movies=[{
    "title": "Avengers: Endgame",
    "genre": "Action/Sci-Fi",
    "description":"After Thanos, an intergalactic warlord, disintegrates half of the universe, the Avengers must reunite and assemble again to reinvigorate their trounced allies and restore balance.",
    "year": 2019,
    "imageUrl": "Endgame",
    "videoId": "hA6hldpSTF8",
    "rating": 4.9
}]

test('Testing the Search Bar', () => {
    const handleChange = jest.fn()
    const {container} = render(<SearchBar changed={handleChange} />)
    const input=container.firstChild.childNodes[1]
    fireEvent.change(input, { target: { value: 'a' } })
    expect(handleChange).toHaveBeenCalledTimes(1)
    expect(input.value).toBe('a')
})

jest.mock('./../molecules/MovieCard', () => ({
    __esModule: true,
    default: () => { return <div></div>}
}))

test("Match the snapshot - MovieList",() => {
    const input=create(<MovieList movies={movies} />);
    expect(input.toJSON()).toMatchSnapshot();
})