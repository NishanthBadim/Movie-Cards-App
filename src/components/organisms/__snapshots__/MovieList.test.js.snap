// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot - MovieList 1`] = `
<div>
  <div
    className="form-group has-search"
  >
    <span
      className="fa fa-search form-control-feedback"
    />
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search by Movie Name..."
      size="40"
      style={
        Object {
          "display": "inline-block",
        }
      }
      type="text"
      value=""
    />
    <p
      className=""
      style={
        Object {
          "color": "gray",
          "display": "inline-block",
          "fontSize": "18px",
        }
      }
    >
      1 Result Found
    </p>
  </div>
  <div
    className="card-deck"
  >
    <div />
  </div>
</div>
`;
